src/main/java/edu/ucsb/cs156/happiercows/controllers/ProfitsController.java

```diff
@@ -0,0 +1,209 @@
+
+        Long userId = getCurrentUser().getUser().getId();
+        UserCommons userCommons = userCommonsRepository.findById(userCommonsId).orElseThrow(() -> new EntityNotFoundException(UserCommons.class, userCommonsId));
+
+        Profit createdProfit = new Profit();
+        createdProfit.setProfit(profit);
+        createdProfit.setUserCommons(userCommons);
+        createdProfit.setTimestamp(timestamp);
+        Profit savedProfit = profitRepository.save(createdProfit);
+        return savedProfit;
+    }
+
+    @ApiOperation(value = "Update a single profit as admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("/admin")
+    public Profit putProfitById_admin(
+            @ApiParam("id") @RequestParam Long id,
+            @RequestBody @Valid Profit newProfit) {
+        Profit profit = profitRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(Profit.class, id));
+
+        profit.setProfit(newProfit.getProfit());
+        profit.setTimestamp(newProfit.getTimestamp());
+
+        profitRepository.save(profit);
+
+        return profit;
+    }
+
+    @ApiOperation(value = "Update a single profit as user")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PutMapping("")
+    public Profit putProfitById(
+            @ApiParam("id") @RequestParam Long id,
+            @RequestBody @Valid Profit newProfit) {
+        Long userId = getCurrentUser().getUser().getId();
+
+        Profit profit = profitRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(Profit.class, id));
+        if (userId != profit.getUserCommons().getUserId())
+            throw new EntityNotFoundException(Profit.class, id);
+
+        profit.setProfit(newProfit.getProfit());
+        profit.setTimestamp(newProfit.getTimestamp());
+
+        profitRepository.save(profit);
+
+        return profit;
+    }
+
+
+    @ApiOperation(value = "Delete other user profit as admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @DeleteMapping("/admin")
+    public Object deleteProfit_Admin(
+            @ApiParam("id") @RequestParam Long id) {
+        Profit profit = profitRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(Profit.class, id));
+
+        profitRepository.delete(profit);
+
+        return genericMessage("Profit with id %s deleted".formatted(id));
+    }
+
+    @ApiOperation(value = "Delete a Profit of the user")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @DeleteMapping("")
+    public Object deleteProfit(
+            @ApiParam("id") @RequestParam Long id) {
+        Long userId = getCurrentUser().getUser().getId();
+
+        Profit profit = profitRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(Profit.class, id));
+
+        if (userId != profit.getUserCommons().getUserId())
+            throw new EntityNotFoundException(Profit.class, id);
+
+        profitRepository.delete(profit);
+
+        return genericMessage("Profit with id %s deleted".formatted(id));
+    }
+}
```

Comment:
What is the user case for a user to delete their own profit?

Shouldnt' this be an admin only function?